chapter05/drjt/ex2.pl

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/perl
+use 5.010;
+use warnings;
+use strict;
+
+# copy-pasted from the chapter (mostly)
+my %total_bytes;
+while ( <> ) {
+  my ( $source, $dest, $bytes ) = split;
+  $total_bytes{$source}{$dest} += $bytes;
+
+  # store the total bytes for each source machine, to sort over later
+  $total_bytes{$source}{total} += $bytes;
+}
+
+# loop over the source machines, sorted by total bytes transfered
+foreach my $source ( sort { $total_bytes{$b}{total} <=> $total_bytes{$a}{total} } keys %total_bytes ) {
+
+  # We don't want to include the total in the output, and we don't need it any more, so remove that from the current source machine's hash here
+  delete $total_bytes{$source}{total};
```

Comment:
This is probably a bit more efficient than the way I chose - my final loop uses a `next if` so has to do a string compare for every destination.  I don't like destroying data :-)